libs/chain-signatures/contract/tests/common.rs

```diff
@@ -512,6 +522,95 @@ pub async fn sign_and_validate(
     Ok(())
 }
 
+pub fn example_secp256k1_point() -> PublicKey {
+    "secp256k1:4Ls3DBDeFDaf5zs2hxTBnJpKnfsnjNahpKU9HwQvij8fTXoCP9y5JQqQpe273WgrKhVVj1EH73t5mMJKDFMsxoEd".parse().unwrap()
+}
+
+// based on https://github.com/near/threshold-signatures/blob/eb04be447bc3385000a71adfcfc930e44819bff1/src/confidential_key_derivation/ckd.rs
+fn hash2curve(app_id: &[u8]) -> ProjectivePoint {
+    const DOMAIN: &[u8] = b"NEAR CURVE_XOF:SHAKE-256_SSWU_RO_";
+    <Secp256k1 as GroupDigest>::hash_from_bytes::<ExpandMsgXof<sha3::Shake256>>(
+        &[app_id],
+        &[DOMAIN],
+    )
+    .unwrap()
+}
+
+/// Derives a confidential key following https://github.com/near/threshold-signatures/blob/main/docs/confidential_key_derivation.md
```

Comment:
nice! I like that you chose to not rely on the threshold-signatures implementation. This should catch regressions!